cmd/snap/cmd_run.go

```diff
@@ -234,7 +234,7 @@ func (x *cmdRun) Execute(args []string) error {
 			return fmt.Errorf(i18n.G("unable to access privileged desktop launcher: unable unable to get session bus: %v"), err)
 		}
 		o := conn.Object("io.snapcraft.Launcher", "/io/snapcraft/PrivilegedDesktopLauncher")
-		call := o.Call("io.snapcraft.PrivilegedDesktopLauncher.OpenDesktopEntry", 0, desktopFile)
+		call := o.Call("io.snapcraft.PrivilegedDesktopLauncher.OpenDesktopEntryWithArguments", 0, desktopFile, args)
```

Comment:
This is a bit too simple, but might be okay for a first pass.

As an example, one of the actions in Firefox's desktop file has:

    Exec=firefox -new-window

In this case, `args == []string{"-new-window"}`, so we need some way to distinguish arguments that are URIs/filenames from other arguments.

Reply:
Added "actions" parameter.